### PR TITLE
[improve][develop-plugin] Add section for AdditionalServlet plugins

### PR DESCRIPTION
**Pull request by alpreu**
## Motivation
There is currently no description of adding custom servlets using Pulsar's plugin architecture

### CODE REVIEW

**Comment by tisonkun** [comment]
I'm not sure who is familiar with plugins, so cc @codelipenghui @merlimat. It's encouraged to delegate to domain experts and we know who is suitable as a reviewer then.

**Comment by alpreu** [comment]
> I'm not sure who is familiar with plugins, so cc @codelipenghui @merlimat. It's encouraged to delegate to domain experts and we know who is suitable as a reviewer then.

I know @andrasbeni wrote an integration test, maybe he can help out with reviewing

File: docs/develop-plugin.md [hunk 1]
```diff
@@ -6,6 +6,125 @@ sidebar_label: "Plugin"
...
+   Create an `additional_servlet.yml` file in the `resources/META-INF/services` directory to describe a NAR file.
+
+   ```conf
+   name: my-servlet
+   description: Describes my-servlet
+   additionalServletClass: org.my.package.MyServlet
+   
+   ```
+
+#### Step 3: package your project into a NAR file
```
Comment (momo-jun) [inline docs/develop-plugin.md:69]:
```suggestion
#### Step 3: Package your project into a NAR file
```

File: docs/develop-plugin.md [hunk 2]
```diff
@@ -6,6 +6,125 @@ sidebar_label: "Plugin"
...
+
+2. Generate a NAR file in the `target` directory.
+
+   ```script
+   
+   mvn clean install
+   
+   ```
+
+#### Step 4: configure and restart broker
```
Comment (momo-jun) [inline docs/develop-plugin.md:107]:
```suggestion
#### Step 4: Configure and restart broker
```

**Comment by andrasbeni** [comment]
The content looks good from an engineering point of view. 
I'd love to see a few words about how `AdditionalServletWithPulsarService` can be used instead of `AdditionalServlet` when the developer wants the servlet to have access to Pulsar internals (for e.g., producing messages or performing administrative tasks). But I don't insist.

File: docs/develop-plugin.md [hunk 3]
```diff
@@ -6,6 +6,125 @@ sidebar_label: "Plugin"
...
+#### Step 1: Create a Maven project
+
+For how to create a Maven project, see [here](https://maven.apache.org/guides/getting-started/maven-in-five-minutes.html).
+
+#### Step 2: Implement the `AdditionalServlet` interface
+
+1. Add a dependency for `pulsar-broker` in the `pom.xml` file as displayed. Otherwise, you can not find the [`AdditionalServlet`](https://github.com/apache/pulsar/blob/master/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/web/plugin/servlet/AdditionalServlet.java) interface.
+
+   ```xml
+
```
Comment (Anonymitaet) [inline docs/develop-plugin.md:38]:
```suggestion
```
Can you remove all blank lines in code blocks (as they are not needed)?

File: docs/develop-plugin.md [hunk 4]
```diff
@@ -6,6 +6,125 @@ sidebar_label: "Plugin"
...
+   ```xml
+   
+   <dependency>
+      <groupId>org.apache.pulsar</groupId>
+      <artifactId>pulsar-broker</artifactId>
+      <version>${pulsar.version}</version>
+      <scope>provided</scope>
+   </dependency>
+
```
Comment (Anonymitaet) [inline docs/develop-plugin.md:45]:
```suggestion
```

File: docs/develop-plugin.md [hunk 5]
```diff
@@ -6,6 +6,125 @@ sidebar_label: "Plugin"
...
+3. Describe a NAR file.
+
+   Create an `additional_servlet.yml` file in the `resources/META-INF/services` directory to describe a NAR file.
+
+   ```conf
+   name: my-servlet
+   description: Describes my-servlet
+   additionalServletClass: org.my.package.MyServlet
+
```
Comment (Anonymitaet) [inline docs/develop-plugin.md:66]:
```suggestion
```

**Review by Anonymitaet — APPROVED** [review]
Thanks for your contribution! Overall LGTM. I left a minor suggestion.

**Comment by alpreu** [comment]
Thanks for the review everyone! I applied your suggestions and added a small not for when to use `AdditionalServletWithPulsarService`

**Review by momo-jun — APPROVED** [review]
@alpreu Thanks for adding the docs!
By the way, 2.11.x was released during the review window. Is it possible that you can also sync the contributions to pulsar-site/versioned_docs/version-2.11.x? Thank you.

**Comment by alpreu** [comment]
@momo-jun I rebased now and added the 2.11 version as well, should be good to go now :)

**Pull request merged**